homeassistant/components/somfy/switch.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Somfy Camera Shutter."""
+from pymfy.api.devices.camera_protect import CameraProtect
+from pymfy.api.devices.category import Category
+
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES, API
+from homeassistant.helpers.entity import ToggleEntity
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy switch platform."""
+
+    def get_shutters():
+        """Retrieve switches."""
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [
+            SomfyCameraShutter(device, hass.data[DOMAIN][API])
+            for device in devices
+            if Category.CAMERA.value in device.categories
+        ]
+
+    async_add_entities(await hass.async_add_executor_job(get_shutters), True)
+
+
+class SomfyCameraShutter(SomfyEntity, ToggleEntity):
```

Comment:
We should inherit `SwitchDevice` imported from the switch base entity integration instead of `ToggleEntity`.

Reply:
That's what I've used for the first time, but SwitchDevice has methods related to electricity. ToggleEntity instead has only the methods I care about.

Reply:
All switch entities should inherit `SwitchDevice`. Not all properties of that class are required to implement.